fivetran_provider_async/operators.py

```diff
@@ -128,35 +70,36 @@ def __init__(
         self.poll_frequency = poll_frequency
         self.schedule_type = schedule_type
         self.reschedule_wait_time = reschedule_wait_time
-        super().__init__(
-            connector_id=self.connector_id,
-            run_name=self.run_name,
-            timeout_seconds=self.timeout_seconds,
-            fivetran_conn_id=self.fivetran_conn_id,
-            fivetran_retry_limit=self.fivetran_retry_limit,
-            fivetran_retry_delay=self.fivetran_retry_delay,
-            poll_frequency=self.poll_frequency,
-            schedule_type=self.schedule_type,
-            **kwargs,
-        )
+        self.deferrable = deferrable
+        super().__init__(**kwargs)
 
-    def execute(self, context: Dict[str, Any]) -> None:
+    def execute(self, context: Dict[str, Any]) -> None | str:
         """Start the sync using synchronous hook"""
         hook = self._get_hook()
         hook.prep_connector(self.connector_id, self.schedule_type)
-        hook.start_fivetran_sync(self.connector_id)
-
-        # Defer and poll the sync status on the Triggerer
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=FivetranTrigger(
-                task_id=self.task_id,
-                fivetran_conn_id=self.fivetran_conn_id,
-                connector_id=self.connector_id,
-                poke_interval=self.poll_frequency,
-                reschedule_wait_time=self.reschedule_wait_time,
-            ),
-            method_name="execute_complete",
+        last_sync = hook.start_fivetran_sync(self.connector_id)
+
+        if not self.deferrable:
+            return last_sync
+        else:
+            # Defer and poll the sync status on the Triggerer
```

Comment:
have an additional check on sync status before deferring. what if the sync is already complete by the time this is invoked

Reply:
I have added a check PTAL